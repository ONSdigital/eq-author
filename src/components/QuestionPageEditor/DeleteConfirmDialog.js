import React from "react";

import styled from "styled-components";

import ModalDialog from "components/ModalDialog";
import DialogHeader from "components/Dialog/DialogHeader";
import DialogIcon from "components/Dialog/DialogIcon";
import {
  Message,
  Heading,
  Subheading,
  Description
} from "components/Dialog/DialogMessage";
import Button from "components/Button/index";
import ButtonGroup from "components/ButtonGroup/index";

import PropTypes from "prop-types";

import iconAlert from "./icon-alert.svg";
import iconPage from "./icon-dialog-page.svg";

const DeleteConfirmModalDialog = styled(ModalDialog)`
  width: 30em;
`;

const Alert = styled.div`
  display: flex;
  align-items: center;
  font-size: 0.8em;
  margin-bottom: 1em;

  &::before {
    content: url(${iconAlert});
    display: inline-block;
    margin: 0 0.5em 0 0;
  }
`;

const DeleteConfirmDialog = ({
  title,
  onDeletePage,
  onClose,
  ...otherProps
}) => (
  <DeleteConfirmModalDialog onClose={onClose} {...otherProps}>
    <DialogHeader>
      <Message>
        <Heading>Delete</Heading>
        <Subheading>{title}</Subheading>
        <Description>
          <Alert>
            All edits, properties and routing settings will also be removed.
          </Alert>
        </Description>
      </Message>
      <DialogIcon icon={iconPage} />
    </DialogHeader>
    <ButtonGroup horizontal align="right">
      <Button secondary onClick={onClose} data-test="btn-cancel">
        Cancel
      </Button>
      <Button tertiary autoFocus onClick={onDeletePage} data-test="btn-delete">
        Delete
      </Button>
    </ButtonGroup>
  </DeleteConfirmModalDialog>
);

DeleteConfirmDialog.propTypes = {
  title: PropTypes.string,
  onDeletePage: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
};

export default DeleteConfirmDialog;
