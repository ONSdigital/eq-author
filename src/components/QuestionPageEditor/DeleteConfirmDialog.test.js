import DeleteConfirmDialog from "./DeleteConfirmDialog";

import React from "react";
import { shallow } from "enzyme";

describe("Question Page Editor", () => {
  let wrapper;

  let mockMutations;
  let page;

  beforeEach(() => {
    mockMutations = {
      onDeletePage: jest.fn(),
      onClose: jest.fn()
    };

    page = {
      __typename: "Page",
      id: "1",
      title: "",
      description: "",
      guidance: ""
    };

    wrapper = shallow(<DeleteConfirmDialog {...mockMutations} page={page} />);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call deletePage handler when delete button is clicked", () => {
    const deleteBtn = wrapper.find("[data-test='btn-delete']");
    deleteBtn.simulate("click");
    expect(mockMutations.onDeletePage).toHaveBeenCalled();
  });

  it("should call close handler when cancel button is clicked", () => {
    const cancelBtn = wrapper.find("[data-test='btn-cancel']");
    cancelBtn.simulate("click");
    expect(mockMutations.onClose).toHaveBeenCalled();
  });
});
