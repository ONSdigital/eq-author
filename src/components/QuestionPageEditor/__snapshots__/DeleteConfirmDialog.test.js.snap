// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Page Editor should render 1`] = `
<DeleteConfirmDialog__DeleteConfirmModalDialog
  data-test="confirm-delete-dialog"
  onClose={[Function]}
  page={
    Object {
      "__typename": "Page",
      "description": "",
      "guidance": "",
      "id": "1",
      "title": "",
    }
  }
>
  <DialogHeader___default>
    <DialogMessage__Message>
      <DialogMessage__Heading>
        Delete
      </DialogMessage__Heading>
      <DialogMessage__Subheading />
      <DialogMessage__Description>
        <DeleteConfirmDialog__Alert>
          All edits, properties and routing settings will also be removed.
        </DeleteConfirmDialog__Alert>
      </DialogMessage__Description>
    </DialogMessage__Message>
    <DialogIcon___default
      icon="icon-dialog-page.svg"
    />
  </DialogHeader___default>
  <ButtonGroup
    align="right"
    horizontal={true}
  >
    <Button
      data-test="btn-cancel"
      onClick={[Function]}
      secondary={true}
    >
      Cancel
    </Button>
    <Button
      autoFocus={true}
      data-test="btn-delete"
      onClick={[Function]}
      tertiary={true}
    >
      Delete
    </Button>
  </ButtonGroup>
</DeleteConfirmDialog__DeleteConfirmModalDialog>
`;
